src/www/ui/api/Controllers/UploadController.php

```diff
@@ -837,4 +837,54 @@ public function getGroupsWithPermissions($request, $response, $args)
     $res["permGroups"] = $finalPermGroups;
     return $response->withJson($res, 200);
   }
+
+  /**
+   * Set the main license for the upload
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setMainLicense($request, $response, $args)
+  {
+    try {
```

Comment:
Please try to limit the scope of this `try/catch` block to find the issue easily.

Reply:
Done

Reply:
I believe it can be further reduced.

Reply:
Please reduce the scope, then we can merge the PR @dushimsam

Reply:
done